### ISSUE TITLE
CAPEV2 Dirty Line Azure Setup / DNS Problem

### ISSUE THREAD

**Issue by WVAuditors** [kevoreilly/CAPEv2#1053]
Follow up to CCCS-Kevin Comment on #1041. 

"Regarding the dirty line, depending on your network security group rules are implemented, this should be as easy as setting the route to internet in routing.conf or specifying route in the task options of a submission." 

I decided to implement dirty line on my Azure CAPEv2 sandbox setup. I am able to IP forward traffic on VNET2_SUB2 (10.14.2.4) which is my CAPEv2 private address. Running TCPDUMP -i eth1 I can see the traffic being forwarded at my CAPE VM. But I can only get it to work with the original sandbox (Cuckoo1) that I create my image from disk from. Before I pull a new version of image from Cuckoo1 my IP settings are IP: 10.14.2.5 Subnet 255.255.255.0 Gateway: 10.14.2.1 DNS: Azure DNS 168.x.x.x address. After creating a new version of image the VM scale set virtual machine grabs this configuration IP: 10.14.2.6 Subnet 255.255.255.0 Gateway: 10.14.2.1 DNS: 10.14.2.4. This seems to be the issue because 10.14.2.4 is the actual CAPE machine on the VNET2_SUB2 subnet on VNET2 and this is not a DNS server. The provisioned VM from image cannot reach the internet. Any idea why my VM scale set is pulling this configuration? I have search all the logs and configs. I can't find anything that would cause this. I greatly appreciate any help you can provide.

**Comment by doomedraven** [comment]
@cccs-kevin do you know about this? you are the uniq person that uses Azure that i know

**Comment by cccs-kevin** [comment]
@WVAuditors Yes I absolutely know why your VMSS is using the that IP for DNS. It's because in the machinery the DNS server is set to the IP of the resultserver (aka the CAPEv2 private address you mentioned): https://github.com/kevoreilly/CAPEv2/blob/294fcdaa5e5640ef33b2cd55c2c7b03b549bf1be/modules/machinery/az.py#L774

I will add a configuration that allows this value to be set :) In the meantime, you can just hardcode the IP you want into that line.

**Comment by doomedraven** [comment]
thank you Kevin

**Comment by WVAuditors** [comment]
I have tried to hardcode the IP for DNS but I did not have any success doing so. No matter what I set my DNS to it still seems to set 10.14.2.4 as my DNS server.  When saying hardcode do you mean set it to the desired DNS on the cuckoo1 VM I pulling the image from disk from? Or do you mean setting it in the machinery az.py script? I think you probably mean the az.py because that is probably what contains the python lang to create the VMSS instance. Besides the VMSS instance not getting internet access it is working great. I seem to have 0 issues with detonation with Azure. My proxmox setup is a little grumpy sometimes although I don't know why. I setup the proxmox home lab to learn about CAPE so I am not really concerned about that one now. I am curious about 1 particular thing about my proxmox homelab setup on CAPE. When I run python3 cleaners.py --clean on that setup analysis processing fails going forward as for Azure when I do the same everything is perfect. I usually just revert to snapshot on my CAPE in proxmox. It clears the past submissions with no issues and starts over doing successful processing at 1 with no errors or failures. Regardless about it doesn't matter I just thought I would ask about that because it has always bugged me. Glad to see im not having that same issue in Azure.

**Comment by doomedraven** [comment]
just update now and set proper value

without error what you getting on homelab i can't help

**Comment by WVAuditors** [comment]
Ok next time I use my proxmox setup I will get an error for you if I can. If i remember correct though it just clears the past submission with no error but will never process another submission. It simply just fails. I appreciate all your help guys you rock!

**Comment by doomedraven** [comment]
you are welcome. always try provide log, that speedup a lot bug finding

**Comment by cccs-kevin** [comment]
@WVAuditors Can you please open a new issue regarding what you are experiencing on your proxmox setup?

And yes by hardcode I meant in az.py. I just created a PR https://github.com/kevoreilly/CAPEv2/pull/1055 that you can implement on your setup which should set the DNS server from the config file. That being said, you may have to destroy the current VMSS you have and recreate it by re-running CAPE for the DNS server config to take effect.

**Comment by WVAuditors** [comment]
Please excuse my ignorance but I ran the command to update but I think it may be pulling from the wrong repo possibly. I don't think it is pulling the correct az.py and az.conf. Do you mind to educate me on the correct way to git from the correct location. It is pulling from master I think I am not up to speed on how to pull from community updates obviously.

**Comment by doomedraven** [comment]
* `cat .git/config` verify that link is poiting to this repo. 
* Config won't be updated due to your changes so you need to create backup as example `mv conf/az.conf conf/az.conf_b`.
* `git pull` so you have latest config. you can restore now config and add new config field by copy new fields by hand

**Comment by WVAuditors** [comment]
Thank you doomedraven & ccs-kevin! I was able to get the dirty line running. Not immediately but I had to mess with the iptables again. I think this is because I had rebooted the CAPE yesterday evening. Thanks again guys you rock! Now I am working Py PIL for the screenshots. 

sudo sysctl -w net.ipv4.ip_forward=1
sudo iptables -A FORWARD -o eth1 -j ACCEPT
sudo iptables -A FORWARD -i eth1 -j ACCEPT
sudo iptables -t nat -A POSTROUTING -o eth1 -j MASQUERADE
